lib/appsignal/check_in.rb

```diff
@@ -2,10 +2,18 @@
 
 module Appsignal
   module CheckIn
+    HEARTBEAT_FOREVER_INTERVAL_SECONDS = 30
     class << self
+      @@forever_heartbeats = [] # rubocop:disable Style/ClassVars
+
+      # @api private
+      def kill_forever_heartbeats
+        @@forever_heartbeats.each(&:kill)
+      end
```

Comment:
I'm not sure why the class env var is needed here.

This will work the way you want it I think, a single list of these forever heartbeats kept on the module.

```suggestion
      # @api private
      def forever_heartbeats
        @forever_heartbeats ||= []
      end

      # @api private
      def kill_forever_heartbeats
        forever_heartbeats.each(&:kill)
      end
```

Reply:
I think I tried this and it didn't work -- I'll double-check. 👍

Reply:
I tried it locally before commenting and it worked for me :)

Reply:
Ah, indeed -- what didn't work was trying to initialise it as `@forever_heartbeats = []` within `class << self` itself.